src/lib/helpers.ts

```diff
@@ -8,3 +21,47 @@ export function getUsdcAddress(networkId: number): string {
 
   return buyTokenAddress;
 }
+
+export async function sendGaslessTransaction(contractCall: PromiEvent<Contract>): Promise<any> {
+  let hashOutcome = OUTCOMES.INITIAL;
+
+  const transactionHash = await new Promise(
+    (resolve, reject) => {
+      contractCall.on('error', (error: Error) => {
+        if (hashOutcome === OUTCOMES.INITIAL) {
+          hashOutcome = OUTCOMES.REJECTED;
+          reject(error);
+          (contractCall as any).off();
+        }
+      });
+
+      contractCall.on('transactionHash', (txHash: string) => {
+        if (hashOutcome === OUTCOMES.INITIAL) {
+          hashOutcome = OUTCOMES.RESOLVED;
+          resolve(txHash);
+          (contractCall as any).off();
+        }
```

Comment:
So this only resolves if `hashOutcome` was not re-assigned in the above callback?

Reply:
Can you add comments describing this? IMO initially it's confusing what the `OUTCOMES` enum is for.